qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java

```diff
@@ -63,9 +63,29 @@ public XrayTestEntity buildTestEntity(XrayTest xrayTest, T scenarioData) {
         entity.setComment(getComment(xrayTest, scenarioData));
         entity.setExecutedBy(getExecutedBy(xrayTest, scenarioData));
         entity.setAssignee(getAssignee(xrayTest, scenarioData));
-        entity.setSteps(getSteps(xrayTest, scenarioData));
-        entity.setExamples(getExamples(xrayTest, scenarioData));
-        entity.setIterations(getIterations(xrayTest, scenarioData));
+
+        // Only when there is one execution we can add steps directly to the test entity
+        if (scenarioData instanceof TestScenarioLogCollection) {
+            entity.setSteps(getSteps(xrayTest, scenarioData));
+        } else {
+            entity.setSteps(null);
+        }
+
+        // Empty arrays for examples are not accepted by the Xray API
+        List<String> examples = getExamples(xrayTest, scenarioData);
+        if (examples != null && examples.size() > 0) {
```

Comment:
`!examples.empty()` is probably cleaner code.

Reply:
Ok, fixed it